text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,236 @@
+3) The [`package.toml`](https://buildpacks.io/docs/reference/package-config/) file should have additional metadata for each dependencies to specify
+    - the endpoint a dependency is exposed at via the lifecycle 
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Answering the following should outline the behavior of this feature.
+
+### How will dependencies appear on the filesystem during the `build` phase?
+
+Dependencies will be listed underneath `/cnb/deps/` with a schema that has the following properties:
+
+- dependency placement does not depend on the dependencies bits (no hashes, SHA256, ...).
+- dependency placement is calculable by the lifecycle (needed to provide through `CNB_DEPS`).
+- dependency placement is specific enough to reduce inter-buildpack collisions.
+- dependency placement provides compartmentalization between buildpacks, (`CNB_DEPS` only provides paths to dependencies a buildpack should have access to).
+
+```
+cnb
+├─ buildpacks
+│   └─ ruby-buildpack
+│       └─ 1.2.3
+│           ├── bin
+│           │   ├── build
+│           │   └── detect
+│           └── buildpack.toml
+├─ deps
+│   ├── io.paketo.mri
+│   │    ├── v2.7.1.toml
+│   │    ├── v2.7.1
+│   │    ├── v2.6.8.toml
+│   │    └── v2.6.8
+│   ├── io.paketo.other-dep
+│   │    ├── v2.2.2.toml
+│   │    └── v2.2.2
+│   └── <id>
+│        ├── v1.2.3.toml
+│        ├── v1.2.3
+│        ├── <version>
+│        └── <version>.toml
+└── lifecycle
+```
+### How will paths to dependencies be exposed to a buildpack?
+
+During the `build` phase of the `paketo-buildpacks/mri` buildpack, the `CNB_DEPS` environment variable will provide a path pointing to a `packaged.toml` file, this will contain information on all all dependencies packaged with a buildpack.
+
+More on the contents of this file [here](#What-are-the-contents-of-the-packagedtoml-file-that-CNB_PATH-points-to).
+
+
+### How are dependencies added to a buildpackage, what is the `pack` user experience?
+These dependencies are added to a buildpackage by specifying fields in the `package.toml`
+
+``` toml
+[buildpack]
+uri = "./path/to/buildpack.tgz"
+
+[[dependencies]]
+uri = "path/to/dependency"
+id = "org.some-identifier"
+version = "v1.2.3"
+  [dependencies.metadata]
+  # dependency-specific data
+
+[[dependencies]]
+uri = "https://buildpacks.cloudfoundry.org/dependencies/node/node_10.20.0_linux_x64_cflinuxfs3_76f4a198.tgz"
+ID = "paketo.node"
+version = "v10.20.0"
+  [dependencies.metadata]
+  cool = "beans"
+
+
+```
+
+the above would give us the following file tree.
+
+```
+cnb
+└─ deps
+    ├── org.some-identifier
+    │    ├── v1.2.3.toml
+    │    └── v1.2.3
+    └── paketo.node
+         ├── v10.20.0.toml
+         └── v10.20.0
+
+```
+
+
+### How are dependencies grouped as layers
+Layers containing dependencies are attached as the final layers to a buildpackage. The exact distribution of dependencies into layers is left up to the packaging utility.
```

Comment:
Would we want this spec'd so the images produced are the same/deterministic?

Reply:
Ya definitely, I'll add a total ordering scheme to these additional layers.